### PR TITLE
feat(transport authority applications): transfer of vehicle ownership

### PR DESCRIPTION
**Pull request by johannaagma**
# ...

New application for Samgöngustofa: Eigendaskipti ökutækis (Transfer of Vehicle Ownership).

Note: We had a bigger PR (https://github.com/island-is/island.is/pull/9046) and decided to split it up, so we can first focus on Transfer of Vehicle Ownership, and then the rest of applications. So this PR might include some code that is not being used (yet), but will be used in the next PR.

## What

Setup application for Transfer of Vehicle Ownership in the application-system.

Note: We also had to change how we send in chargeItemCode into PaymentController (apps/application-system/api/src/app/modules/payment/payment.controller.ts), because we need the possibility to send inn multiple chargeItemCodes.

## Why

Move application from "Mínar síður Samgöngustofa" to island.is.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by sigruntg — APPROVED** [review]

**Review by brynjarorng — APPROVED** [review]
LGTM(Devops)

File: libs/clients/transport-authority/vehicle-codetables/src/lib/vehicleCodetablesClient.config.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+export const VehicleCodetablesClientConfig = defineConfig<
+  z.infer<typeof schema>
+>({
+  name: 'VehicleCodetablesClient',
+  schema,
+  load(env) {
+    return {
+      xroadBaseUrl: env.required('XROAD_BASE_PATH', 'http://localhost:8081'),
+      xroadClientId: env.required(
```
Comment (eirikurn) [inline libs/clients/transport-authority/vehicle-codetables/src/lib/vehicleCodetablesClient.config.ts:18]: Fyi. We have a shared config file for these two. Can check some of the other clients how we use it.

File: libs/clients/transport-authority/vehicle-owner-change/src/lib/vehicleOwnerChangeClient.service.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  // Note: auth.nationalId might not be the same as ownerChange.seller.ssn, but when calling this
+  // function, we should be sure that the seller (ownerChange.seller.ssn) is the one that created
+  // the application and is the only one that has changed the fields: permno, seller and buyer
+  public async saveOwnerChange(
+    auth: User,
+    ownerChange: OwnerChange,
+  ): Promise<void> {
+    await this.ownerchangeApiWithAuth(auth).rootPost({
+      apiVersion: '2.0',
+      apiVersion2: '2.0',
```
Comment (eirikurn) [inline libs/clients/transport-authority/vehicle-owner-change/src/lib/vehicleOwnerChangeClient.service.ts:47]: Interesting :)

File: libs/api/domains/transport-authority/transfer-of-vehicle-ownership/src/lib/graphql/main.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { Query, Resolver } from '@nestjs/graphql'
+import { ApiScope } from '@island.is/auth/scopes'
+import { UseGuards } from '@nestjs/common'
+import { IdsUserGuard, Scopes, ScopesGuard } from '@island.is/auth-nest-tools'
+import { InsuranceCompany } from './models'
+import { TransferOfVehicleOwnershipApi } from '../transferOfVehicleOwnership.service'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(ApiScope.internal)
+@Resolver()
```
Comment (eirikurn) [inline libs/api/domains/transport-authority/transfer-of-vehicle-ownership/src/lib/graphql/main.resolver.ts:10]: We might want to change this to be a "public" resolver (no scope check) so it works for different delegations later on.

**Review by disaerna — APPROVED** [review]
Code owned files are 👍

**Review by eirikurn — APPROVED** [review]
LGTM

File: apps/application-system/api/src/app/modules/payment/payment.controller.ts [hunk 4]
```diff
@@ -72,13 +72,18 @@ export class PaymentController {
...
+        0,
+      ),
       definition: {
-        chargeItemName: chargeItem.chargeItemName,
-        chargeItemCode: chargeItem.chargeItemCode,
-        performingOrganiationID: chargeItem.performingOrgID,
-        chargeType: chargeItem.chargeType,
-        amount: chargeItem.priceAmount,
+        performingOrganizationID: chargeItems[0].performingOrgID,
+        chargeType: chargeItems[0].chargeType,
```
Comment (kksteini) [inline apps/application-system/api/src/app/modules/payment/payment.controller.ts:81]: Will all the items have non-differing `performingID`s and `chargeType`s?

**Review by kksteini — APPROVED** [review]
Codeowner Files LGTM

**Review by obmagnusson — APPROVED** [review]
CodeOwner files LGTM

**Pull request merged**